igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/InfoController.groovy

```diff
@@ -59,44 +65,27 @@ class InfoController {
     WerckerProperties werckerProperties
 
     @RequestMapping(value = '/masters', method = RequestMethod.GET)
-    List<Object> listMasters(
-        @RequestParam(value = "showUrl", defaultValue = "false") String showUrl,
-        @RequestParam(value = "type", defaultValue = "") String type) {
-
-        BuildServiceProvider providerType = (type == "") ? null :
-            BuildServiceProvider.valueOf(type.toUpperCase())
-
-        if (showUrl == 'true') {
-            List<Object> masterList = []
-            addMaster(masterList, providerType, jenkinsProperties,  BuildServiceProvider.JENKINS)
-            addMaster(masterList, providerType, travisProperties,   BuildServiceProvider.TRAVIS)
-            addMaster(masterList, providerType, gitlabCiProperties, BuildServiceProvider.GITLAB_CI)
-            addMaster(masterList, providerType, werckerProperties,  BuildServiceProvider.WERCKER)
-            return masterList
+    @PostFilter("hasPermission(filterObject, 'BUILD_SERVICE', 'READ')")
+    List<String> listMasters(@RequestParam(value = "type", defaultValue = "") String type) {
+
+        BuildServiceProvider providerType = (type == "") ? null : BuildServiceProvider.valueOf(type.toUpperCase())
+        //Filter by provider type if it is specified
+        if (providerType) {
+            return buildServices.getServiceNames(providerType)
         } else {
-            //Filter by provider type if it is specified
-            if (providerType) {
-                return buildServices.getServiceNames(providerType)
-            } else {
-                return buildServices.getServiceNames()
-            }
+            return buildServices.getServiceNames()
         }
     }
 
-    void addMaster(masterList, providerType, properties, expctedType) {
-        if (!providerType || providerType == expctedType) {
-            masterList.addAll(
-                properties?.masters.collect {
-                    [
-                        "name"         : it.name,
-                        "address"      : it.address
-                    ]
-                }
-            )
+    @RequestMapping(value = '/buildServices', method = RequestMethod.GET)
+    List<BuildServiceView> getAllBuildServices() {
```

Comment:
Can the logic to construct this be moved to `BuildServices.java`?  I just refactored a lot of logic out of the controllers, trying to push them into Java classes, and would like to avoid moving more logic back to the controllers.